.circleci/config.yml

```diff
@@ -561,6 +565,49 @@ jobs:
           event: fail
           template: basic_fail_1 # see https://github.com/CircleCI-Public/slack-orb/wiki#templates.
 
+  run_and_convert_notebooks_to_docs:
+    docker:
+      - image: 307629990626.dkr.ecr.us-east-1.amazonaws.com/deci/infra/circleci/runner/sg-gpu:<< pipeline.parameters.sg_docker_version >>
+    resource_class: deci-ai/sg-gpu-on-premise
+    parameters:
+      sg_existing_env_path:
+        type: string
+        default: "/env/persistent_env"
+      sg_new_env_name:
+        type: string
+        default: "${CIRCLE_BUILD_NUM}"
+      sg_new_env_python_version:
+        type: string
+        default: "python3.8"
+    steps:
+      - checkout
+      - run:
+          name: install requirements and execute notebooks
+          command: |
+            << parameters.sg_new_env_python_version >> -m venv << parameters.sg_new_env_name >>
+            source << parameters.sg_new_env_name >>/bin/activate
+            python3.8 -m pip install --upgrade setuptools pip wheel
+            python3.8 -m pip install -r requirements.txt
+            python3.8 -m pip install -r requirements.dev.txt
+            python3.8 -m pip install .
+            python3.8 -m pip install torch torchvision torchaudio
+            python3.8 -m pip install pytorch-quantization==2.1.2 --extra-index-url https://pypi.ngc.nvidia.com
```

Comment:
Do you think these should already be there?
The notebooks have specifc installations executed in them.

Reply:
That's good point. Probably we should not install anything here and really rely on notebook itself to install what it needs. This way we will ensure it is self-contained.
Ideally we would have to have fresh env per notebook, but that would be quite hard to achieve.

Reply:
Ok I think I found the solution.

1) Added a new Makefile target `check_notebooks_version_match` that would go over notebooks we want to export and check whether version of SG installed in them matches with the version of SG defined in __version__ variable (See https://github.com/Deci-AI/super-gradients/blob/b03485c01a071aa0561ea68f09578487ddc83776/tests/verify_notebook_version.py). This check guarantees that we ALWAYS have matching versions between SG.__version__ and notebooks

2) We want to keep `python3.8 -m pip install .` here, since it would allow us to install any feature/tag version of SG. 

3) Now the magic part. So if we have SG.__version__==3.2.1 and our notebook has any other version it will immediately quit with non-zero error code and build will be marked as failed. This would prevent us from releasing a version if there is version mismatch

4) If SG.__version__ matches the one in notebook then notebook executed as usual.

So I think this pretty much solves the problem of outdated notebooks during release. This however does not solve the problem of notebooks execution result not being comited to git but that's a different story.

Reply:
Nice one!